src/biogeophys/BalanceCheckMod.F90

```diff
@@ -624,124 +610,115 @@ subroutine BalanceCheck( bounds, &
+       errseb_max_val = maxval( abs(errseb(bounds%begp:bounds%endp)))
+       write(iulog,*)'errseb_max_val      :', errseb_max_val
+
+       if ((errseb_max_val > energy_warning_thresh) .and. (DAnstep > skip_steps)) then
+
+           indexp = maxloc( abs(errseb(bounds%begp:bounds%endp)), 1 ) + bounds%begp -1
+           indexc = patch%column(indexp)
+           indexg = patch%gridcell(indexp)
+
+           write(iulog,*)'WARNING: BalanceCheck: surface flux energy balance error (W/m2)'
+           write(iulog,*)'nstep          = ' ,nstep
+           write(iulog,*)'errseb         = ' ,errseb(indexp)
+
+           if ( errseb_max_val > error_thresh ) then
+              write(iulog,*)'clm model is stopping - error is greater than 1e-5 (W/m2)'
+              write(iulog,*)'sabv           = ' ,sabv(indexp)
+              write(iulog,*)'sabg           = ' ,sabg(indexp), ((1._r8- frac_sno(indexc))*sabg_soil(indexp) + &
+                   frac_sno(indexc)*sabg_snow(indexp)),sabg_chk(indexp)
+              write(iulog,*)'forc_tot      = '  ,forc_solad(indexg,1) + forc_solad(indexg,2) + &
+                   forc_solai(indexg,1) + forc_solai(indexg,2)
+
+              write(iulog,*)'eflx_lwrad_net = ' ,eflx_lwrad_net(indexp)
+              write(iulog,*)'eflx_sh_tot    = ' ,eflx_sh_tot(indexp)
+              write(iulog,*)'eflx_lh_tot    = ' ,eflx_lh_tot(indexp)
+              write(iulog,*)'eflx_soil_grnd = ' ,eflx_soil_grnd(indexp)
+              write(iulog,*)'fsa fsr = '        ,fsa(indexp),    fsr(indexp)
+              write(iulog,*)'fabd fabi = '      ,fabd(indexp,:), fabi(indexp,:)
+              write(iulog,*)'albd albi = '      ,albd(indexp,:), albi(indexp,:)
+              write(iulog,*)'ftii ftdd ftid = ' ,ftii(indexp,:), ftdd(indexp,:),ftid(indexp,:)
+              write(iulog,*)'elai esai = '      ,elai(indexp),   esai(indexp)
+              write(iulog,*)'clm model is stopping'
+              call endrun(decomp_index=indexp, clmlevel=namep, msg=errmsg(sourcefile, __LINE__))
+           end if
+
        end if
 
        ! Soil energy balance check
 
-       found = .false.
-       do c = bounds%begc,bounds%endc
-          if (col%active(c)) then
-             if (abs(errsoi_col(c)) > 1.0e-5_r8 ) then
-                found = .true.
-                indexc = c
-             end if
-          end if
-       end do
-       if ( found ) then
-          write(iulog,*)'WARNING: BalanceCheck: soil balance error (W/m2)'
-          write(iulog,*)'nstep         = ',nstep
-          write(iulog,*)'errsoi_col    = ',errsoi_col(indexc)
-          if (abs(errsoi_col(indexc)) > 1.e-4_r8 .and. (DAnstep > skip_steps) ) then
-             write(iulog,*)'clm model is stopping'
-             call endrun(decomp_index=indexc, clmlevel=namec, msg=errmsg(sourcefile, __LINE__))
-          end if
-       end if
-
+       errsoi_col_max_val  =  maxval( abs(errsoi_col(bounds%begc:bounds%endc)) , mask = (errsoi_col(bounds%begc:bounds%endc) /= spval) )
```

Comment:
Mostly as a reminder for myself: We talked about changing this to use a mask of col%active rather than spval here, as well as a couple of lines down.

Reply:
Correct! It uses `col%active` as the mask for `errsoi_col now`.

Reply:
For some reason, using `col%active` as the mask for `errsoi_col` still gives out 1E+036 for maximum `errsoi_col` ( model stops) for the following six cases:

```
ERP_Ly3_P72x2.f10_f10_musgs.IHistClm50BgcCrop.cheyenne_intel.clm-cropMonthOutput
ERP_P36x2_D_Ld10.f10_f10_musgs.IHistClm50SpG.cheyenne_intel.clm-glcMEC_decrease
ERP_P36x2_D_Ld5.f10_f10_musgs.IHistClm45BgcCruGs.cheyenne_intel.clm-decStart
ERP_P36x2_Lm13.f10_f10_musgs.IHistClm50Bgc.cheyenne_intel.clm-monthly
ERS_Ly3_P72x2.f10_f10_musgs.IHistClm50BgcCropG.cheyenne_intel.clm-cropMonthOutput
ERP_P36x2_Lm13.f10_f10_musgs.IHistClm50Bgc.cheyenne_gnu.clm-monthly
```

They are all `P36x2` or `P72x2` and I think it might be some issue related to the threading.

Reply:
It looks like the problem is that you masked by `col%active`, without including bounds. Just like the other arrays, this column-level array needs explicit bounds, so the masking should be done by `col%active(bounds%begc:bounds%endc)`. Can you try making that change and rerunning those 6 tests? If things look good for those 6 tests, I'd be comfortable with your making the tag without needing to rerun the full test suite, if that would save you time. (We could just call Sam's attention to this, so that, in the unlikely event that he runs into balance check errors in his testing, he can come back to us.)

Reply:
Thanks for your comment. It is corrected. I reran the full test suites and generated the baselines for this tag. Should I go ahead for the updates on the logs?  
I'd be happy to incorporate the changes in `run_sys_tests` in the PR, but there was one issue using gnu compiler, which I wanted to discuss with you.